tests/integration/targets/dcnm_res_manager/tests/dcnm/dcnm_res_manager_query.yaml

```diff
@@ -0,0 +1,302 @@
+
+    - assert:
+        that:
+          - '(result["response"] | length) != 0'
+
+    - name: Query Resources - filter by switch and pool name
+      cisco.dcnm.dcnm_res_manager: 
+        state: query                                 # choose form [merged, deleted, query]
+        fabric: "{{ ansible_it_fabric }}" 
+        config:
+          - pool_name: "BGP_ASN_ID"                  # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"       
+          - pool_name: "L3_VNI"                      # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"        
+          - pool_name: "L2_VNI"                      # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"         
+          - pool_name: "LOOPBACK_ID"                 # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"          
+          - pool_name: "VPC_PEER_LINK_VLAN"          # Based on the 'poolType', select appropriate name 
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"           
+          - pool_name: "VPC_ID"                      # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"            
+          - pool_name: "LOOPBACK0_IP_POOL"           # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"             
+          - pool_name: "LOOPBACK1_IP_POOL"           # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"              
+          - pool_name: "SUBNET"                      # Based on the 'poolType', select appropriate name
+            switch:                                  # provide the switch information to which the given resource is attached
+              - "{{ ansible_switch1 }}"
+      register: result
+
+    - assert:
+        that:
+          - '(result["response"] | length) != 0'
```

Comment:
Should we be checking for specific information in the query result?

Reply:
I can check for the requested resources. but in case it is a query all, we will not know what all resources will be returned. So there is no way for us to check for specific names. But for specific test cases requesting for specific resources, we can check.